cli/js/lib.deno.ns.d.ts

```diff
@@ -32,12 +33,16 @@ declare namespace Deno {
    * when `Deno.runTests` is used */
   export function test(name: string, fn: TestFunction): void;
 
+  enum TestStatus {
+    Passed = "passed",
+    Failed = "failed",
+    Skipped = "skipped"
+  }
```

Comment:
We should probably collapse all of little interfaces and enums for testing under a single namespace...

Reply:
Just realised these are all implicitly public via `TestReporter` in `RunTestsOptions` so that would be necessary. The `TestEvent*` stuff can and should be hidden IMO with some restructuring. All that needs to be public is `TestReporter`, `TestResult` and `TestStats` each with no references to any other proprietary type.

Reply:
How is this?
```ts
namespace tests {
  export interface Result {
    name: string;
    status: "passed" | "failed" | "skipped";
    duration?: number;
    error?: Error;
  }

  export interface Stats {
    filtered: number;
    ignored: number;
    measured: number;
    passed: number;
    failed: number;
  }

  export interface StartEvent {
    tests: Deno.TestDefinition[];
  }

  export interface ResultEvent {
    result: Result;
  }

  export interface EndEvent {
    stats: Stats;
    duration: number;
    results: Result[];
  }

  export interface Reporter {
    start(msg: StartEvent): Promise<void>;
    result(msg: ResultEvent): Promise<void>;
    end(msg: EndEvent): Promise<void>;
  }

  export class ConsoleReporter implements Reporter {
    // ...
  }
}
```

Reply:
@nayeemrmn I already needed to split "result" into two separate messages for https://github.com/denoland/deno/pull/4371
We should figure this out before 1.0, but it's not actionable for this PR